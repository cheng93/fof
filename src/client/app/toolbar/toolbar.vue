<template>
    <header class="mdc-top-app-bar mdc-top-app-bar--fixed">
        <div class="mdc-top-app-bar__row">
            <div class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start">
                <span class="mdc-top-app-bar__title">
                    <router-link class="link mdc-theme--text-primary-on-primary" to="/">
                        {{title}}
                    </router-link>
                </span>
            </div>
        </div>
    </header>
</template>

<script>
export default {
    props: {
        title: {
            type: String,
            required: true
        }
    }
};
</script>
