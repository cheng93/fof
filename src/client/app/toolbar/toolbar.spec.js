import { mount, RouterLinkStub } from '@vue/test-utils';
import toolbar from './toolbar.vue';

describe('Toolbar', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(toolbar, {
            propsData: {
                title: 'Hello World'
            },
            stubs: {
                RouterLink: RouterLinkStub
            }
        });
    });
    it('matches snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should link to home', () => {
        expect(wrapper.find(RouterLinkStub).props().to).toBe('/');
    });
});
