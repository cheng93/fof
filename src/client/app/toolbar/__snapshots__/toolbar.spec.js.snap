// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar matches snapshot 1`] = `
<header class="mdc-top-app-bar mdc-top-app-bar--fixed">
  <div class="mdc-top-app-bar__row">
    <div class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"><span class="mdc-top-app-bar__title"><a class="link mdc-theme--text-primary-on-primary">
                    Hello World
                </a></span></div>
  </div>
</header>
`;
